src/crimson/mon/MonClient.cc

```diff
@@ -65,25 +83,35 @@ class Connection {
 
 private:
   seastar::future<> setup_session(epoch_t epoch,
-				  const AuthMethodList& auth_methods,
-				  const EntityName& name);
-  std::unique_ptr<AuthClientHandler> create_auth(Ref<MAuthReply> m,
-						 const EntityName& name,
-						 uint32_t want_keys);
+                                  const EntityName& name);
+  std::unique_ptr<AuthClientHandler> create_auth(ceph::auth::method_t,
+                                                 uint64_t global_id,
+                                                 const EntityName& name,
+                                                 uint32_t want_keys);
   seastar::future<bool> do_auth();
 
 private:
   bool closed = false;
+  // v1
   seastar::promise<Ref<MAuthReply>> reply;
+  // v2
+  using clock_t = seastar::lowres_system_clock;
+  clock_t::time_point auth_start;
+  ceph::auth::method_t auth_method = 0;
+  seastar::promise<> auth_done;
+  const AuthRegistry& auth_registry;
```

Comment:
nit: `::setup_session()` makes use of the `auth_registry` while it can be called also from `::authenticate_v1`. Because of that I believe it might reasonable to move this 1 line down, under the `// v1 and v2` comment.

Reply:
fixed and repushed.